gravitee-apim-console-webui/src/services-ngx/environment-api-headers.service.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+import { Inject, Injectable } from "@angular/core";
+import { HttpClient } from "@angular/common/http";
+import { BehaviorSubject, EMPTY, Observable } from "rxjs";
+import { catchError, switchMap, take, tap } from "rxjs/operators";
+
+import { SnackBarService } from "./snack-bar.service";
+
+import { Constants } from "../entities/Constants";
+import { ApiPortalHeader } from "../entities/apiPortalHeader";
+
+
+@Injectable({
+  providedIn: "root"
+})
+export class EnvironmentApiHeadersService {
+  private headersList$$: BehaviorSubject<ApiPortalHeader[]> = new BehaviorSubject([]);
+
+  constructor(
+    @Inject("Constants") private readonly constants: Constants,
+    private httpClient: HttpClient,
+    private snackBarService: SnackBarService,
+  ) {
+  }
+
+  public getHeadersList$(): Observable<ApiPortalHeader[]> {
+    return this.headersList$$.asObservable();
+  }
+
+  getApiHeaders(): Observable<ApiPortalHeader[]> {
+    return this.httpClient
+      .get<ApiPortalHeader[]>(`${this.constants.env.baseURL}/configuration/apiheaders/`)
+      .pipe(
+        tap((apiHeadersRes: ApiPortalHeader[]) => {
+          this.headersList$$.next(apiHeadersRes);
+        }),
+        catchError(_ => {
+          this.snackBarService.error('Error occurred.');
+          return [];
+        })
+      );
+  }
+
+  createApiHeader(apiHeader: ApiPortalHeader): Observable<any> {
```

Comment:
```suggestion
  createApiHeader(apiHeader: ApiPortalHeader): Observable<void> {
```
We try to avoid any in our code base. And to type correctly. 

For the moment there is too much legacy but soon we will be able to activate a stricter lint or tsconfig on this.

Reply:
resolved